### ISSUE TITLE
kernel 6.1.53-v8+ broke official 7'' touchscreen

### ISSUE THREAD

**Issue by rafael2k** [raspberrypi/linux#5619]
### Describe the bug

The touchscreen was working flawless for years... until kernel 6.1.53 broke it.

 [    8.914075] edt_ft5x06 10-0038: supply vcc not found, using dummy regulator
[    8.914404] edt_ft5x06 10-0038: supply iovcc not found, using dummy regulator
[    8.915006] edt_ft5x06 10-0038: touchscreen probe failed
[    8.915382] edt_ft5x06: probe of 10-0038 failed with error -5

More reports at:
https://forums.raspberrypi.com/viewtopic.php?t=356352

### Steps to reproduce the behaviour

Just use a kernel  6.1.53 or newer ( 6.1.54 has the same problem).
Using Raspberry OS Bullseye 64 bit here. 

### Device (s)

Raspberry Pi 4 Mod. B

### System

Raspberry Pi reference 2023-05-03
Generated using pi-gen, https://github.com/RPi-Distro/pi-gen, 7c750947a959fb626a70c09fd17c65815df192ac, stage2

Sep 21 2023 18:42:00 
Copyright (c) 2012 Broadcom
version 3094eda5aadf2c719edd2c68d1900884abc3dbca (clean) (release) (start)

Linux blablabla 6.1.54-v8+ #1682 SMP PREEMPT Wed Sep 20 15:19:57 BST 2023 aarch64 GNU/Linux

### Logs

 [    8.914075] edt_ft5x06 10-0038: supply vcc not found, using dummy regulator
[    8.914404] edt_ft5x06 10-0038: supply iovcc not found, using dummy regulator
[    8.915006] edt_ft5x06 10-0038: touchscreen probe failed
[    8.915382] edt_ft5x06: probe of 10-0038 failed with error -5

### Additional context

-

**Comment by pelwell** [comment]
Do you have a Last Known Good kernel version?

**Comment by rafael2k** [comment]
I was using something like 6.1.3x, and it was all good. I can try to pinpoint at least the minor version which broke. Do you happen to know how to use rpi-update to make it install older kernel versions?
Of course I can just compile the kernel... but then I'll take much more time.

**Comment by pelwell** [comment]
I see that 6.1.21 is working - I'll bisect from there.

**Comment by rafael2k** [comment]
6.1.21 is definitely working, indeed, as available via the apt repository (from Raspberry OS) or from the official Raspberry OS images.

**Comment by rafael2k** [comment]
for the reference, my config.txt:
```
dtparam=i2c_arm=on
dtparam=i2s=on
dtparam=audio=off
display_auto_detect=1
dtoverlay=vc4-kms-v3d,noaudio
max_framebuffers=2
arm_64bit=1
disable_overscan=1
disable_splash=1
otg_mode=1

[pi4]
arm_boost=1
# this is for my RTC and bit-banged i2c I use
dtoverlay=i2c-rtc-gpio,ds1307,i2c_gpio_delay_us=10,bus=2,i2c_gpio_sda=13,i2c_gpio_scl=6
# my soundcard
dtoverlay=audioinjector-wm8731-audio
dtoverlay=i2s-mmap
avoid_warnings=1
disable_poe_fan=1
force_eeprom_read=0
dtoverlay=disable-bt
```

**Comment by pelwell** [comment]
This is caused by vc4-kms-dsi-7inch overlay, which got damaged between 6.1.47 and 6.1.48, when the support for i2c_csi_dsi0 was added, by the addition of another level of fragment merging.

Fragments should be thought of as being merged in order, where the order is the order in which they appear in the file, not by the numerical ordering of the fragments. However, the firmware and other overlay tools have an additional rule saying that fragments that patch things within the overlay ("intra" fragments) should be merged before fragments which patch the base DTB ("extra" fragments). What they can't do is detect and correctly order a chain of patches.

In this case, fragment 5 patched fragment 12, fragment 12 patched fragment 13, and fragment 13 patched the base DTB - which sounds OK written like that, but not when fragment 5 appears after fragment 13 in the file. At the point the overlay is applied, the intra fragments get merged first, with 12 patching 13 and 5 patching 12 _in that order_, so the contributions from fragment 5 - `vcc-supply` and `reset-gpio` - get lost, disabling the touchscreen by keeping it reset.

Combining fragments 12 and 13 removes the extra level of dependency, and the disabling of fragment 12 can be replaced by setting its `status` property to `disabled`.

#5622 is my proposed solution. If you wait about an hour until the build checks complete you'll be able to run `sudo rpi-update pulls/5622` to confirm that it fixes the problem.

**Comment by pelwell** [comment]
[ Previous description updated to correct a few errors ]

**Comment by rafael2k** [comment]
I will test ASAP. Thanks a lot!

**Comment by rafael2k** [comment]
Tested and confirmed it works! Yay!

**Comment by pelwell** [comment]
Thanks - that patch will be in all future kernel builds.

**Comment by Hooba97** [comment]
Hey , thank you for following the issue, been waiting for over a week for a solution, can you help me how to apply the fix please

**Comment by pelwell** [comment]
Apply it to what? If you are building your own kernel then just clone this repo and you'll get the fix. If you just want to install a kernel with the fix run `sudo rpi-update pulls/5622`. If you don't have rpi-update, first run `git clone https://github.com/raspberrypi/rpi-update.git` and get it from there.

**Comment by Hooba97** [comment]
Thank you @pelwell , touch screen and camera v3 are working perfectly now , and thank you @rafael2k for opening the case here to solve the problem

**Comment by rafael2k** [comment]
Yay! @pelwell, can we close the issue?

**Comment by hydazz** [comment]
still experiencing this on `6.1.58-v8`, also tried `rpi-update pulls/5622` (back to `6.1.54-v8+`), same error:

```
[   11.133848] edt_ft5x06 10-0038: touchscreen probe failed
[   11.134136] edt_ft5x06: probe of 10-0038 failed with error -5
```
RPI4B, Bookworm (PIOS Lite) 64Bit

edit:
Thought i was going crazy, but now crazier, reverted back to 6.1.21. still failing to probe touchscreen:

```
[   10.280537] edt_ft5x06 10-0038: touchscreen probe failed
[   10.280777] edt_ft5x06: probe of 10-0038 failed with error -121
```
different error message though.

I tested all the pins on the screen to the pi and they all had continuity...

**Comment by pelwell** [comment]
After running `rpi-update pulls/5622` again (and rebooting, of course), what do the following commands report?:
```
$ dmesg | grep edt_ft5x06`
$ ls -l /proc/device-tree/soc/i2c0mux/i2c@1/ts@38
```

**Comment by hydazz** [comment]
> After running `rpi-update pulls/5622` again (and rebooting, of course), what do the following commands report?:
> 
> ```
> $ dmesg | grep edt_ft5x06`
> $ ls -l /proc/device-tree/soc/i2c0mux/i2c@1/ts@38
> ```

```
alex@livingroom-dashboard:~ $ uname -r
6.1.54-v8+
alex@livingroom-dashboard:~ $ dmesg | grep edt_ft5x06
[   10.054085] edt_ft5x06 10-0038: supply iovcc not found, using dummy regulator
[   10.397893] edt_ft5x06 10-0038: touchscreen probe failed
[   10.398163] edt_ft5x06: probe of 10-0038 failed with error -5
alex@livingroom-dashboard:~ $ ls -l /proc/device-tree/soc/i2c0mux/i2c@1/ts@38
total 0
-r--r--r-- 1 root root 15 Oct 26 20:18 compatible
-r--r--r-- 1 root root  3 Oct 26 20:18 name
-r--r--r-- 1 root root  4 Oct 26 20:18 phandle
-r--r--r-- 1 root root  4 Oct 26 20:18 reg
-r--r--r-- 1 root root 12 Oct 26 20:18 reset-gpio
-r--r--r-- 1 root root  0 Oct 26 20:18 touchscreen-inverted-x
-r--r--r-- 1 root root  0 Oct 26 20:18 touchscreen-inverted-y
-r--r--r-- 1 root root  4 Oct 26 20:18 touchscreen-size-x
-r--r--r-- 1 root root  4 Oct 26 20:18 touchscreen-size-y
-r--r--r-- 1 root root  4 Oct 26 20:18 vcc-supply
alex@livingroom-dashboard:~ $
```

**Comment by pelwell** [comment]
Thanks - to me that shows that the overlay is doing what is should (both vcc-supply and reset-gpio are present). The fact that you are getting errors on the latest(ish) kernel and 6.1.21 suggest, while for other people it is working, suggests a hardware/wiring fault in your setup.

**Comment by hydazz** [comment]
> Thanks - to me that shows that the overlay is doing what is should (both vcc-supply and reset-gpio are present). The fact that you are getting errors on the latest(ish) kernel and 6.1.21 suggest, while for other people it is working, suggests a hardware/wiring fault in your setup.



does `failed with error -5` point to anything?

**Comment by pelwell** [comment]
> does failed with error -5 point to anything?

    #define	EIO		 5	/* I/O error */

Yes - an I2C access is failing, i.e. a hardware problem. Error -121 is similar:

    #define	EREMOTEIO	121	/* Remote I/O error */

**Comment by PinkFreud** [comment]
> Thanks - to me that shows that the overlay is doing what is should (both vcc-supply and reset-gpio are present). The fact that you are getting errors on the latest(ish) kernel and 6.1.21 suggest, while for other people it is working, suggests a hardware/wiring fault in your setup.

I'm chiming in with another unresponsive touchscreen.  Same errors for me:
```
[   10.540015] edt_ft5x06 10-0038: supply vcc not found, using dummy regulator
[   10.540342] edt_ft5x06 10-0038: supply iovcc not found, using dummy regulator
[   10.541029] edt_ft5x06 10-0038: touchscreen probe failed
[   10.541435] edt_ft5x06: probe of 10-0038 failed with error -5
```

This Pi was running the bullseye image up until yesterday, and the touchscreen worked without issue.  Once I booted bookworm, the touchscreen was unresponsive, and all attempts to revive it, including the aforementioned `rpi-update pulls/5622` fail for me - rpi-update succeeds, but a reboot doesn't bring the touchscreen back to life.

Edit: no vcc-supply or reset-gpio here for me.
```
pi@schwartz:~ $ ls -al /proc/device-tree/soc/i2c0mux/i2c@1/ts@38
total 0
drwxr-xr-x 2 root root  0 Oct 26 22:58 .
drwxr-xr-x 6 root root  0 Oct 26 22:21 ..
-r--r--r-- 1 root root 15 Oct 26 22:58 compatible
-r--r--r-- 1 root root  3 Oct 26 22:58 name
-r--r--r-- 1 root root  4 Oct 26 22:58 phandle
-r--r--r-- 1 root root  4 Oct 26 22:58 reg
-r--r--r-- 1 root root  0 Oct 26 22:58 touchscreen-inverted-x
-r--r--r-- 1 root root  0 Oct 26 22:58 touchscreen-inverted-y
-r--r--r-- 1 root root  4 Oct 26 22:58 touchscreen-size-x
-r--r--r-- 1 root root  4 Oct 26 22:58 touchscreen-size-y
pi@schwartz:~ $ 
```

**Comment by pelwell** [comment]
I just tried this with a 32-bit Bookworm image running the latest stable kernel on a 4B:
```
pi@raspberrypi:~$ uname -a
Linux raspberrypi 6.1.58-v7l+ #1694 SMP Mon Oct 30 11:51:22 GMT 2023 armv7l GNU/Linux
```
Attaching only the display ribbon, 5V and GND, and making no configuration changes, it just works - I can click, double-click and drag. Relevant messages in the kernel log are:
```
[   11.679899] edt_ft5x06 10-0038: supply iovcc not found, using dummy regulator
[   12.004473] input: generic ft5x06 (00) as /devices/platform/soc/fe205000.i2c/i2c-22/i2c-10/10-0038/input/input2
```
`config.txt` includes `dtoverlay=vc4-kms-v3d` and `disable_fw_kms_setup=1` (removing the latter doesn't stop it working).

Switching the same image and display over to a 3B+ I get the same result (just slower), although of course it's running the pre-Wayland lxpanel desktop:
```
[   15.761012] edt_ft5x06 10-0038: supply iovcc not found, using dummy regulator
[   16.365789] input: generic ft5x06 (00) as /devices/platform/soc/3f205000.i2c/i2c-11/i2c-10/10-0038/input/input1
```

**Comment by pelwell** [comment]
(To install the latest stable kernel, run `sudo rpi-update stable`, having first backed up any important data)

**Comment by MartinHajda** [comment]
Thanks to @rafael2k for the thread and @pelwell for fixing the problem, the display now works as well as before. Good work!

**Comment by pelwell** [comment]
And hopefully this kind of error will be caught by the improved overlay checks.

**Comment by barajaam** [comment]
Touch Screen on 6.1.69-v8+ #1711 not working any help is appreciated.
Tried pretty much all the recommended solutions on this page but none of them worked

**Comment by ForbiddenEra** [comment]
Not sure if I'm running into this, maybe..?

On cm4io, trying to get touchscreen working..

```
forbiddenera@io:~ $ i2cdetect -y 0
     0  1  2  3  4  5  6  7  8  9  a  b  c  d  e  f
00:                         -- -- -- -- -- -- -- --
10: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
20: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
30: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
40: -- -- -- -- -- 45 -- -- -- -- -- -- -- -- -- --
50: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
60: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
70: -- -- -- -- -- -- -- --
forbiddenera@io:~ $ ls -l /proc/device-tree/soc/i2c0mux/i2c@1/ts@38
total 0
-r--r--r-- 1 root root 15 Aug 27 06:30 compatible
-r--r--r-- 1 root root  3 Aug 27 06:30 name
-r--r--r-- 1 root root  4 Aug 27 06:30 phandle
-r--r--r-- 1 root root  4 Aug 27 06:30 reg
-r--r--r-- 1 root root 12 Aug 27 06:30 reset-gpio
-r--r--r-- 1 root root  0 Aug 27 06:30 touchscreen-inverted-x
-r--r--r-- 1 root root  0 Aug 27 06:30 touchscreen-inverted-y
-r--r--r-- 1 root root  4 Aug 27 06:30 touchscreen-size-x
-r--r--r-- 1 root root  4 Aug 27 06:30 touchscreen-size-y
-r--r--r-- 1 root root  4 Aug 27 06:30 vcc-supply
```

when on ds0, with the jumpers set.. same result with no jumpers set and gpio 0/1 connected to sda/scl on the touchscreen..

verified cable works with pi5, screen works fine with the adapter cable on the 5, and directly connected to a 3b, etc..

tried ds1 too of course - ds1 was my default, even..

in `dmesg`:
```
forbiddenera@io:~ $ dmesg | grep edt_ft5x06
[    4.796293] edt_ft5x06 10-0038: supply iovcc not found, using dummy regulator
[    5.120106] edt_ft5x06 10-0038: touchscreen probe failed
[    5.121440] edt_ft5x06 10-0038: probe with driver edt_ft5x06 failed with error -5
```

```
forbiddenera@io:~ $ uname -a
Linux io 6.12.34+rpt-rpi-v8 #1 SMP PREEMPT Debian 1:6.12.34-1+rpt1~bookworm (2025-06-26) aarch64 GNU/Linux
```

Added `dtoverlay=vc4-kms-dsi-7inch` as per the directions at https://www.raspberrypi.com/documentation/computers/compute-module.html#connect-a-display-to-disp0dsi0

**Comment by 6by9** [comment]
> Not sure if I'm running into this, maybe..?

Seeing as this issue was relating to 6.1.53 and you're running 6.12.34, no you're not hitting the same issue.

Please raise a new issue, and don't truncate or filter logs as you'll often end up hiding the critical bit.

**Comment by ForbiddenEra** [comment]
> > Not sure if I'm running into this, maybe..?
> 
> Seeing as this issue was relating to 6.1.53 and you're running 6.12.34, no you're not hitting the same issue.

Said like regressions don't happen ever.

> and don't truncate or filter logs as you'll often end up hiding the critical bit.

Literally provided logs as per the debugging in the thread.

**Comment by pelwell** [comment]
> truncate
> verb
> 1.
> shorten the duration or extent of.

> filter
> verb
> 1.
> process or assess (items) in order to reject those that are unwanted.

**Comment by 6by9** [comment]
> > > Not sure if I'm running into this, maybe..?
> > 
> > 
> > Seeing as this issue was relating to 6.1.53 and you're running 6.12.34, no you're not hitting the same issue.
> 
> Said like regressions don't happen ever.

It's far more likely that it is something new, and it's easier to flag something as a duplicate than to try and untangle two unrelated issues.

> > and don't truncate or filter logs as you'll often end up hiding the critical bit.
> 
> Literally provided logs as per the debugging in the thread.

That was the guidance given in this thread as something specific was requested.
I'm asking you to open a new thread, and to not truncate or filter the logs.

You also have discussion on a [forum thread](https://forums.raspberrypi.com/viewtopic.php?p=2334288)
